src/cookies.ts

```diff
@@ -0,0 +1,161 @@
+const memoizedCookies: Map<string, string> = new Map<string, string>();
+
+export const getCookieValues = (name: string): string[] => {
+	const nameEq = `${name}=`;
+	const cookies = document.cookie.split(';');
+
+	return cookies.reduce((acc: string[], cookie: string) => {
+		const cookieTrimmed: string = cookie.trim();
+		if (cookieTrimmed.startsWith(nameEq)) {
+			acc.push(
+				cookieTrimmed.substring(nameEq.length, cookieTrimmed.length),
+			);
+		}
+
+		return acc;
+	}, []);
+};
+
+// subset of https://github.com/guzzle/guzzle/pull/1131
+const isValidCookieValue = (name: string) =>
+	!/[()<>@,;"\\/[\]?={} \t]/g.test(name);
```

Comment:
Is it worth extracting this Regex to a constant, as we've done for the `validCountryCode`?

Reply:
good spot